crates/kitsune_p2p/kitsune_p2p/src/types/agent_store.rs

```diff
@@ -73,18 +75,28 @@ impl AgentInfoSigned {
     serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, derive_more::AsRef, Hash, Eq,
 )]
 pub struct AgentInfo {
-    // The space this agent info is relevant to.
+    /// The space this agent info is relevant to.
     space: KitsuneSpace,
-    // The pub key of the agent id this info is relevant to.
+    /// The pub key of the agent id this info is relevant to.
     agent: KitsuneAgent,
-    // List of urls the agent can be reached at, in the agent's own preference order.
+    /// List of urls the agent can be reached at, in the agent's own preference order.
     urls: Urls,
-    // The unix ms timestamp that the agent info was signed at, according to the agent's own clock.
+    /// The unix ms timestamp that the agent info was signed at, according to the agent's own clock.
     #[as_ref(ignore)]
     signed_at_ms: u64,
-    // The expiry ttl for the agent info relative to the signing time.
+    /// The expiry ttl for the agent info relative to the signing time.
     #[as_ref(ignore)]
     expires_after_ms: u64,
+    /// Extra info as encoded MessagePack data.
+    #[serde(with = "serde_bytes")]
+    meta_info: Vec<u8>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, Hash, Eq)]
+/// Extra info that is not used by the bootstrap server.
+pub struct AgentMetaInfo {
+    /// The half length of the [`DhtArc`]
+    pub dht_arc_half_length: u32,
```

Comment:
There's the potential to have multiple arcs published with AgentInfo eventually - we might want to identify this as `dht_storage_arc_half_length`?